exalearn-ui/src/app/guards/role-coach.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs';
+import { UserService } from '../services/user.service';
+
+@Injectable({
+	providedIn: 'root'
+})
+export class RoleCoachGuard implements CanActivate {
+	constructor(private userService: UserService, private router: Router) {}
+
+	canActivate(
+		route: ActivatedRouteSnapshot,
+		state: RouterStateSnapshot
+	): Observable<boolean> | Promise<boolean> | boolean {
+		if (`${this.userService.currentUser?.role.toLocaleLowerCase()}` !== 'coach') {
```

Comment:
Don't we have a enum with roles?
If we don't it makes sense to create one and make sure we use it everywhere.

Reply:
Done, also used new enum in RoleService